### ISSUE TITLE
[no-var] var & Type-checking for globalThis

### ISSUE THREAD

**Issue by maxmellen** [typescript-eslint/typescript-eslint#3617]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ]
}
```

```TS
declare global {
  // Unexpected var, use let or const instead. eslint(no-var)
  declare var myGlobalVar: Foo
}

```

**Expected Result**

Since TypeScript 3.4, declaring a variable in the _global_ context using the `var` keyword adds this variable as a member of `globalThis` (and thus `window`) whereas using `let` or `const` does not ([see here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#type-checking-for-globalthis))

Therefore, I would expect the recommended ruleset to disable eslint's `no-var` warning when inside of a global declare block

**Actual Result**

Currently, using this plugin, ESLint doesn't understand that there's any difference to using `var` in that context and fires the `no-var` warning indiscriminately

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.1` |
| `@typescript-eslint/parser`        | `4.28.1` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.28.0` |
| `node`                             | `14.17.2` |

**Comment by bradzacher** [comment]
If you want a var for its semantics.. then use a var.
If you want a const/let for its semantics.. then use a const/let.

It's impossible for the lint rule to understand your exact intention here.
This is what disable comments are for "I am purposely circumventing the lint rule in this case for reason X".
